### PR TITLE
My code review 2 suggestions

### PR DESCRIPTION
**Pull request by brownadam010**
1. This program allows for an individual to put in a drink and track how many drinks they've had at that point in time while also telling them a rough estimate of their intoxication level. With tweaks I'm sure this app could become a very spot on representation of a persons intoxication level. 
2. yes the program was available on time.
3. There is minimal documentation but each method is placed in way that all logic is very easy to step through and quite intuitive. 
4. yes the program compiles. 
5. I didn't change as much as I added to the program. At this point in time they did not have any live data coming through yet so I went ahead and added the RetrofitClientInstance and set up essentially everything they would need to start testing live data coming through their JSON stream. I added the class that would test live data, not just Mockk, so that when they get their input stream in and going they can start writing unit tests quickly. I also added a mockk test to test gibberish data because that was not included yet. 

Three Technical
1. Enhancing knowledge on unit tests by creating a unit test that incorporates two mockk searches and tests for gibberish at the same time. 
2. Strengthened knowledge on how to get live data through the system by creating the structure of the RetrofitClientInstanct object.
3. Learned where the JSON stream comes in by setting up the drink DAO to allow for outside data to be used in an application.

### CODE REVIEW

File: app/build.gradle
Comment (brownadam010) [inline app/build.gradle:29]: Allows for new version of retrofit to be used without throwing an error.

File: app/build.gradle
Comment (brownadam010) [inline app/build.gradle:45]: implementing retrofit imports.

File: app/src/main/java/com/example/KnowYouLimit/RetrofitClientInstance.kt
Comment (brownadam010) [inline app/src/main/java/com/example/KnowYouLimit/RetrofitClientInstance.kt:21]: Adding retrofitclientinstance object to be used with JSON stream base_url

File: app/src/main/java/com/example/KnowYouLimit/dao/IDrinksDao.kt
Comment (brownadam010) [inline app/src/main/java/com/example/KnowYouLimit/dao/IDrinksDao.kt:15]: allowing for JSON data link to be inserted and used in program.

File: app/src/test/java/com/example/KnowYouLimit/DrinksDataUnitClass.kt
Comment (brownadam010) [inline app/src/test/java/com/example/KnowYouLimit/DrinksDataUnitClass.kt:5]: live data unit test class creation for when JSON link is inserted.

File: app/src/test/java/com/example/KnowYouLimit/DrinksMockkUnitTest.kt
Comment (brownadam010) [inline app/src/test/java/com/example/KnowYouLimit/DrinksMockkUnitTest.kt:97]: gibberish test.

File: app/src/test/java/com/example/KnowYouLimit/DrinksMockkUnitTest.kt
Comment (brownadam010) [inline app/src/test/java/com/example/KnowYouLimit/DrinksMockkUnitTest.kt:21]: changing data unit test to mockk because this is only testing mockk cases.

File: app/src/test/java/com/example/KnowYouLimit/DrinksMockkUnitTest.kt
Comment (brownadam010) [inline app/src/test/java/com/example/KnowYouLimit/DrinksMockkUnitTest.kt:49]: adding other drink case.

File: app/src/test/java/com/example/KnowYouLimit/DrinksMockkUnitTest.kt
Comment (brownadam010) [inline app/src/test/java/com/example/KnowYouLimit/DrinksMockkUnitTest.kt:55]: changing mockk test to test for those two cases and other cases return nothing.

File: app/src/main/java/com/example/KnowYouLimit/RetrofitClientInstance.kt
Comment (discospiff) [inline app/src/main/java/com/example/KnowYouLimit/RetrofitClientInstance.kt:8]: The BASE_URL will need to be populated for this to be used.

File: app/src/main/java/com/example/KnowYouLimit/dao/IDrinksDao.kt
Comment (discospiff) [inline app/src/main/java/com/example/KnowYouLimit/dao/IDrinksDao.kt:10]: The BASE_URL will need to be populated for this to be used.  Where is it going to get the data?

File: app/src/test/java/com/example/KnowYouLimit/DrinksDataUnitClass.kt
Comment (discospiff) [reply app/src/test/java/com/example/KnowYouLimit/DrinksDataUnitClass.kt:5]: I caution against adding an empty class.  If this never gets used, it's clutter, which is technical debt.  Only add this class when it will be used.

**Comment by discospiff** [comment]
This is a new feature, which is not complete, so it's not in a state where it can be merged to master.  What I recommend:
1) The product owner should determine if this is a feature aligns to the product backlog.  If not, delete this branch; don't merge.
2) If the product owner decides it does provide value, then leave this branch open, complete the implementation in this branch, and merge once this feature meets the project's Definition of Done.

**Comment by discospiff** [comment]
I recommend removing the commented out code, as this reviewer has done.  That's easy to do manually.

Naming conventions and other changes... I'd likely leave them as they are.  The team can make a final call on this.

**Comment by discospiff** [comment]
It's not easy to tell what was changed here... as it appears that MainActivity.kt was deleted in one package, and then created in another.  Without enough detail on the change, I don't recommend merging.

**Review by mastroja — APPROVED** [review]

**Pull request merged**
